ddtrace/tracer/tracer.go

```diff
@@ -148,15 +164,18 @@ func newUnstartedTracer(opts ...StartOption) *tracer {
 	} else {
 		writer = newAgentTraceWriter(c, sampler)
 	}
-	return &tracer{
+	t := &tracer{
 		config:           c,
 		traceWriter:      writer,
 		out:              make(chan []*span, payloadQueueSize),
 		stop:             make(chan struct{}),
 		rulesSampling:    newRulesSampler(c.samplingRules),
 		prioritySampling: sampler,
 		pid:              strconv.Itoa(os.Getpid()),
+		features:         &agentFeatures{},
 	}
+	t.stats = newConcentrator(c)
```

Comment:
nit: why don't we create the stats directly in the struct? (and not modify the return)

Reply:
Good point. It was a left-over from a previous form where the `newConcentrator` func was taking the tracer as an argument.